src/components/GenericPageHeader.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <div :class="headerClasses">
+    <s-button v-if="hasButtonBack" type="action" size="medium" icon="chevron-left-rounded" @click="handleBack" />
+    <h3 class="page-header-title">{{ title }}</h3>
+    <s-tooltip
+      v-if="!!tooltip"
+      class="page-header-tooltip"
+      popper-class="info-tooltip info-tooltip--page-header"
+      :content="tooltip"
+      theme="light"
+      :placement="tooltipPlacement"
+      animation="none"
+      :show-arrow="false"
+    >
+      <s-icon name="info" size="16" />
+    </s-tooltip>
+    <!-- TODO: Add appropriate icon -->
+    <s-button v-if="hasButtonHistory" class="el-button--history" type="action" size="medium" icon="time" @click="handleViewTransactionsHistory" />
+    <slot />
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop } from 'vue-property-decorator'
+import TranslationMixin from '@/components/mixins/TranslationMixin'
+import router from '@/router'
+import { PageNames } from '@/consts'
+
+@Component
+export default class GenericPageHeader extends Mixins(TranslationMixin) {
+  @Prop({ default: false, type: Boolean }) readonly hasButtonBack!: boolean
+  @Prop({ default: PageNames.Pool, type: String }) readonly backPageName!: string
+  @Prop({ default: false, type: Boolean }) readonly hasButtonHistory!: boolean
+  @Prop({ default: '', type: String }) readonly title!: string
+  @Prop({ default: '', type: String }) readonly tooltip!: string
+  @Prop({ default: 'bottom-end', type: String }) readonly tooltipPlacement!: string
+
+  get headerClasses (): string {
+    const baseClass = 'page-header'
+    const classes = [baseClass]
+
+    if (this.hasButtonBack) {
+      classes.push(`${baseClass}--center`)
+    }
+
+    return classes.join(' ')
+  }
+
+  handleBack (): void {
+    router.push({ name: this.backPageName })
+  }
+
+  handleViewTransactionsHistory (): void {
```

Comment:
Why generic page header has `PageNames.BridgeTransactionsHistory`? I guess that this is not really generic part. 

Maybe we can replace boolean prop with `historyType` with values like "all" | "bridge" | "". Empty value will mean `hasButtonHistory=false`

Reply:
I'll fix it in another way.